h/static/scripts/group-forms/components/test/CreateEditGroupForm-test.js

```diff
@@ -235,6 +257,118 @@ describe('CreateGroupForm', () => {
+    beforeEach(() => {
+      config.context.group = {
+        pubid: 'testid',
+        name: 'Test Name',
+        description: 'Test group description',
+        link: 'https://example.com/groups/testid',
+      };
+      config.api.updateGroup = {
+        method: 'PATCH',
+        url: 'https://example.com/api/group/foo',
+      };
+    });
+
+    it('displays an edit-group form', async () => {
+      const { wrapper, elements } = createWrapper();
+      const headerEl = elements.header.element;
+      const nameEl = elements.name.fieldEl;
+      const descriptionEl = elements.description.fieldEl;
+      const submitButtonEl = elements.submitButton.element;
+
+      assert.equal(headerEl.text(), 'Edit group');
+      assert.equal(nameEl.getDOMNode().value, config.context.group.name);
+      assert.equal(
+        descriptionEl.getDOMNode().value,
+        config.context.group.description,
+      );
+      assert.equal(submitButtonEl.text(), 'Save changes');
+      assert.isTrue(wrapper.exists('[data-testid="back-link"]'));
+      assert.isFalse(wrapper.exists('[data-testid="error-message"]'));
+      await assertInLoadingState(wrapper, false);
+      assert.isFalse(wrapper.exists('[data-testid="check"]'));
+    });
+
+    it('updates the group', async () => {
+      const { wrapper, elements } = createWrapper();
+      const nameEl = elements.name.fieldEl;
+      const descriptionEl = elements.description.fieldEl;
+
+      const name = 'Edited Group Name';
+      const description = 'Edited group description';
+      nameEl.getDOMNode().value = name;
+      nameEl.simulate('input');
+      descriptionEl.getDOMNode().value = description;
+      descriptionEl.simulate('input');
+      await wrapper.find('form[data-testid="form"]').simulate('submit');
+
+      assert.isTrue(
+        fakeCallAPI.calledOnceWithExactly(
+          config.api.updateGroup.url,
+          config.api.updateGroup.method,
+          {
+            id: config.context.group.pubid,
+            name,
+            description,
+          },
+        ),
+      );
+    });
+
+    it('shows a loading state when the update-group API request is in-flight', async () => {
+      const { wrapper } = createWrapper();
+      fakeCallAPI.resolves(new Promise(() => {}));
+
+      wrapper.find('form[data-testid="form"]').simulate('submit');
+
+      await assertInLoadingState(wrapper, true);
+    });
+
+    it('shows a confirmation after receiving a successful API response', async () => {
+      const { wrapper } = createWrapper();
+      fakeCallAPI.resolves();
+
+      await wrapper.find('form[data-testid="form"]').simulate('submit');
+
+      await assertInLoadingState(wrapper, false);
+      assert.isTrue(wrapper.exists('[data-testid="check"]'));
+    });
+
+    it('shows an error message if callAPI() throws an error', async () => {
+      const errorMessageFromCallAPI = 'Bad API call.';
+      fakeCallAPI.rejects({ message: errorMessageFromCallAPI });
```

Comment:
```suggestion
      fakeCallAPI.rejects(new Error(errorMessageFromCallAPI ));
```

As a matter of habit, it is a good idea to make sure promises always reject with instances of `Error`. In various contexts you get more useful output if this is the case.

Reply:
Done